lib/chromic_pdf/pdf/session.ex

```diff
@@ -21,25 +23,67 @@ defmodule ChromicPDF.Session do
 
   @impl GenServer
   def init(opts) do
-    browser = Keyword.fetch!(opts, :chromic)
+    {:ok, start_session(opts)}
+  end
 
-    {:ok, session_id} =
-      Browser.run_protocol(
-        browser,
-        SpawnSession,
-        Keyword.put_new(opts, :offline, true)
-      )
+  @impl GenServer
+  def handle_call({:run_protocol, protocol_mod, params}, _from, state) do
+    %{opts: opts, session_id: session_id} = state
 
-    {:ok, %{session_id: session_id, browser: browser}}
+    response =
+      session_id
+      |> protocol_mod.new(params)
+      |> run_protocol_through_browser(opts)
+
+    case increase_session_age(state) do
+      {true, new_state} ->
+        {:reply, response, new_state, {:continue, :restart_session}}
+
+      {false, new_state} ->
+        {:reply, response, new_state}
+    end
   end
 
   @impl GenServer
-  def handle_call({:run_protocol, protocol_mod, params}, _from, state) do
-    %{browser: browser, session_id: session_id} = state
+  def handle_continue(:restart_session, state) do
+    %{opts: opts, target_id: target_id} = state
+
+    {:ok, true} =
+      [targetId: target_id]
+      |> CloseTarget.new()
+      |> run_protocol_through_browser(opts)
+
+    {:noreply, start_session(opts)}
+  end
+
+  defp start_session(opts) do
+    {:ok, target_id} = run_protocol_through_browser(CreateTarget.new(), opts)
+
+    {:ok, session_id} =
+      opts
+      |> Keyword.put(:targetId, target_id)
+      |> Keyword.put_new(:offline, true)
+      |> SpawnSession.new()
+      |> run_protocol_through_browser(opts)
+
+    %{
+      opts: opts,
+      target_id: target_id,
+      session_id: session_id,
+      session_age: 0
+    }
+  end
+
+  defp run_protocol_through_browser(protocol, opts) do
+    Browser.run_protocol(Keyword.fetch!(opts, :chromic), protocol)
+  end
+
+  defp increase_session_age(state) do
+    %{opts: opts, session_age: old_session_age} = state
 
-    protocol = protocol_mod.new(session_id, params)
-    response = Browser.run_protocol(browser, protocol)
+    session_age = old_session_age + 1
+    max_session_age = Keyword.get(opts, :max_session_age, @default_max_session_age)
 
-    {:reply, response, state}
+    {session_age >= max_session_age, %{state | session_age: session_age}}
```

Comment:
age sounds like time to me. Maybe `max_session_uses`?

Reply:
or `max_session_commands`?

Reply:
`max_session_prints` :D ?

Reply:
I like `max_session_uses` 👍